src/v/debug_bundle/metadata.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/bytes.h"
+#include "container/fragmented_vector.h"
+#include "debug_bundle/types.h"
+#include "serde/envelope.h"
+#include "serde/rw/envelope.h"
+#include "serde/rw/named_type.h"
+#include "serde/rw/uuid.h"
+#include "serde/rw/variant.h"
+#include "utils/uuid.h"
+
+#include <seastar/core/sstring.hh>
+#include <seastar/util/process.hh>
+#include <seastar/util/variant_utils.hh>
+
+#include <chrono>
+#include <variant>
+
+namespace debug_bundle {
+
+struct process_wait_exited
+  : serde::envelope<
+      process_wait_exited,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    int exit_code;
+
+    auto serde_fields() { return std::tie(exit_code); }
+
+    friend auto
+    operator<=>(const process_wait_exited&, const process_wait_exited&)
+      = default;
+};
+
+struct process_wait_signaled
+  : serde::envelope<
+      process_wait_signaled,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    int signal;
+
+    auto serde_fields() { return std::tie(signal); }
+
+    friend auto
+    operator<=>(const process_wait_signaled&, const process_wait_signaled&)
+      = default;
+};
+
+using serde_process_result
+  = serde::variant<process_wait_exited, process_wait_signaled>;
```

Comment:
note/fyi: these are sort of a pain to change later in my experience, requiring careful handling in serde_read/serde_write. just something to consider.

Reply:
fair, but I think the likelihood of this changing is low (dependent on SS changing the result of `wait()`).  Also processes only really exit with an exit code or if they were signaled.

Reply:
ya, that's what I figured. this seems like a good use case, but as an extension point I've found it to be a source of moderate suffering.

Reply:
> but as an extension point I've found it to be a source of moderate suffering.

sounds like a future mike issue ;)